diff -Nrub --exclude='*~' linux-2.6.21-ccm2200/drivers/usb/net/rndis_host.c linux-2.6.21-rndis-active-sync/drivers/usb/net/rndis_host.c
--- linux-2.6.21-ccm2200/drivers/usb/net/rndis_host.c	2007-06-19 17:22:16.000000000 +0200
+++ linux-2.6.21-rndis-active-sync/drivers/usb/net/rndis_host.c	2007-06-19 18:52:00.000000000 +0200
@@ -462,6 +462,13 @@
 	u.get->len = ccpu2(48);
 	u.get->offset = ccpu2(20);
 
+	/* sleep a little bit until Windows Mobile Device is ready to answer
+	 * our command.
+	 * Some devices (tested HP iPAQ rx1950) will not response to
+	 * the ethernet MAC address command if we don't wait here.
+	 */
+	msleep(150);
+
 	retval = rndis_command(dev, u.header);
 	if (unlikely(retval < 0)) {
 		dev_err(&intf->dev, "rndis get ethaddr, %d\n", retval);
