--- mgetty-1.1.35/Makefile	2006-02-22 14:58:33.000000000 +0100
+++ Makefile	2006-09-22 07:06:50.000000000 +0200
@@ -4,7 +4,7 @@
 #
 # this is the C compiler to use (on SunOS, the standard "cc" does not
 # grok my code, so please use gcc there. On ISC 4.0, use "icc".).
-CC=gcc
+#CC=gcc
 #CC=cc
 #
 #### C Compiler Flags ####
@@ -192,7 +192,7 @@
 #
 # where the configuration files (*.config, aliases, fax.allow/deny) go to
 #
-CONFDIR=$(prefix)/etc/mgetty+sendfax
+CONFDIR=/etc/mgetty
 #CONFDIR=/etc/default/
 #
 #
