diff -Nru '--exclude=*~' buildroot-2.0-org/Makefile buildroot-2.0/Makefile
--- buildroot-2.0-org/Makefile	2008-11-07 09:57:59.000000000 +0100
+++ buildroot-2.0/Makefile	2011-08-01 14:45:53.000000000 +0200
@@ -254,6 +254,7 @@
 endif
 
 include package/*/*.mk
+include swarco/Makefile
 
 TARGETS+=target-devfiles
 
diff -Nru '--exclude=*~' buildroot-2.0-org/package/busybox/Config.in buildroot-2.0/package/busybox/Config.in
--- buildroot-2.0-org/package/busybox/Config.in	2008-11-29 21:12:17.000000000 +0100
+++ buildroot-2.0/package/busybox/Config.in	2011-08-01 14:45:53.000000000 +0200
@@ -51,6 +51,9 @@
 	config BR2_BUSYBOX_VERSION_1_13_X
 		bool "BusyBox 1.13.x"
 
+	config BR2_BUSYBOX_VERSION_1_18_X
+		bool "BusyBox 1.18.x"
+
 	config BR2_PACKAGE_BUSYBOX_SNAPSHOT
 		bool "daily snapshot"
 
@@ -67,6 +70,7 @@
 	default "1.11.3"	if BR2_BUSYBOX_VERSION_1_11_X
 	default "1.12.3"	if BR2_BUSYBOX_VERSION_1_12_X
 	default "1.13.1"	if BR2_BUSYBOX_VERSION_1_13_X
+	default "1.18.5"	if BR2_BUSYBOX_VERSION_1_18_X
 
 config BR2_PACKAGE_BUSYBOX_FULLINSTALL
 	bool "Run BusyBox's own full installation"
diff -Nru '--exclude=*~' buildroot-2.0-org/package/Config.in buildroot-2.0/package/Config.in
--- buildroot-2.0-org/package/Config.in	2008-12-11 01:33:30.000000000 +0100
+++ buildroot-2.0/package/Config.in	2011-08-01 14:45:53.000000000 +0200
@@ -430,4 +430,6 @@
 # various games packages
 source "package/games/Config.in"
 
+source "swarco/package/Config.in"
+
 endmenu
diff -Nru '--exclude=*~' buildroot-2.0-org/package/lighttpd/lighttpd.mk buildroot-2.0/package/lighttpd/lighttpd.mk
--- buildroot-2.0-org/package/lighttpd/lighttpd.mk	2008-12-08 09:15:27.000000000 +0100
+++ buildroot-2.0/package/lighttpd/lighttpd.mk	2011-08-01 14:45:53.000000000 +0200
@@ -8,7 +8,7 @@
 LIGHTTPD_SITE:=http://www.lighttpd.net/download
 LIGHTTPD_INSTALL_STAGING = NO
 LIGHTTPD_INSTALL_TARGET = YES
-LIGHTTPD_DEPENDENCIES = uclibc
+LIGHTTPD_DEPENDENCIES = uclibc zlib
 LIGHTTPD_CONF_ENV =
 
 ifneq ($(BR2_LARGEFILE),y)
@@ -20,6 +20,7 @@
 	--libexecdir=/usr/lib \
 	--localstatedir=/var \
 	--program-prefix="" \
+	--without-bzip2 \
 	$(DISABLE_IPV6) \
 	$(LIGHTTPD_LFS)
 
diff -Nru '--exclude=*~' buildroot-2.0-org/package/mtd/mtd-utils/mtd.mk buildroot-2.0/package/mtd/mtd-utils/mtd.mk
--- buildroot-2.0-org/package/mtd/mtd-utils/mtd.mk	2008-12-08 09:15:27.000000000 +0100
+++ buildroot-2.0/package/mtd/mtd-utils/mtd.mk	2011-08-01 14:45:53.000000000 +0200
@@ -38,6 +38,7 @@
 	CC="$(HOSTCC)" CROSS= CFLAGS=-I$(LINUX_HEADERS_DIR)/include \
 		$(MAKE) LINUXDIR=$(LINUX_DIR) \
 		BUILDDIR=$(MTD_HOST_DIR) \
+		WITHOUT_XATTR=1 \
 		-C $(MTD_HOST_DIR) mkfs.jffs2
 
 $(SUMTOOL): $(MTD_HOST_DIR)/.unpacked
diff -Nru '--exclude=*~' buildroot-2.0-org/package/ntp/ntp.mk buildroot-2.0/package/ntp/ntp.mk
--- buildroot-2.0-org/package/ntp/ntp.mk	2008-12-08 09:15:27.000000000 +0100
+++ buildroot-2.0/package/ntp/ntp.mk	2011-08-01 14:45:53.000000000 +0200
@@ -37,6 +37,9 @@
 		--target=$(GNU_TARGET_NAME) \
 		--host=$(GNU_TARGET_NAME) \
 		--build=$(GNU_HOST_NAME) \
+		--enable-MEINBERG \
+		--enable-RAWDCF \
+		--enable-parse-clocks \
 		--prefix=/usr \
 		--exec-prefix=/usr \
 		--bindir=/usr/bin \
diff -Nru '--exclude=*~' buildroot-2.0-org/package/openssh/openssh.mk buildroot-2.0/package/openssh/openssh.mk
--- buildroot-2.0-org/package/openssh/openssh.mk	2008-09-24 14:00:57.000000000 +0200
+++ buildroot-2.0/package/openssh/openssh.mk	2011-08-01 14:45:53.000000000 +0200
@@ -3,7 +3,7 @@
 # openssh
 #
 #############################################################
-OPENSSH_VERSION=5.1p1
+OPENSSH_VERSION=5.8p2
 OPENSSH_SITE=ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable
 
 OPENSSH_CONF_ENV = LD=$(TARGET_CC)
diff -Nru '--exclude=*~' buildroot-2.0-org/package/openssl/openssl.mk buildroot-2.0/package/openssl/openssl.mk
--- buildroot-2.0-org/package/openssl/openssl.mk	2008-10-13 10:10:35.000000000 +0200
+++ buildroot-2.0/package/openssl/openssl.mk	2011-08-01 14:45:53.000000000 +0200
@@ -3,7 +3,7 @@
 # openssl
 #
 #############################################################
-OPENSSL_VERSION:=0.9.8g
+OPENSSL_VERSION:=1.0.0d
 OPENSSL_SITE:=http://www.openssl.org/source
 
 OPENSSL_TARGET_ARCH=generic32
diff -Nru '--exclude=*~' buildroot-2.0-org/package/pppd/pppd.mk buildroot-2.0/package/pppd/pppd.mk
--- buildroot-2.0-org/package/pppd/pppd.mk	2008-12-08 09:15:27.000000000 +0100
+++ buildroot-2.0/package/pppd/pppd.mk	2011-08-01 13:28:50.000000000 +0200
@@ -3,8 +3,8 @@
 # pppd
 #
 #############################################################
-PPPD_VERSION:=2.4.4
-PPPD_SOURCE:=ppp-$(PPPD_VERSION).tar.gz
+PPPD_VERSION:=2.4.5
+PPPD_SOURCE:=ppp_$(PPPD_VERSION).orig.tar.gz
 PPPD_SITE:=ftp://ftp.samba.org/pub/ppp
 PPPD_DIR:=$(BUILD_DIR)/ppp-$(PPPD_VERSION)
 PPPD_CAT:=$(ZCAT)
@@ -21,11 +21,16 @@
 	$(PPPD_CAT) $(DL_DIR)/$(PPPD_SOURCE) | tar -C $(BUILD_DIR) $(TAR_OPTIONS) -
 	$(SED) 's/ -DIPX_CHANGE -DHAVE_MMAP//' $(PPPD_DIR)/pppd/Makefile.linux
 	$(SED) 's/HAVE_MULTILINK=y/#HAVE_MULTILINK=y/' $(PPPD_DIR)/pppd/Makefile.linux
+	# 2011-08-01 gc: disable plugins
+	$(SED) 's/PLUGIN=y/#PLUGIN=y/' $(PPPD_DIR)/pppd/Makefile.linux
+	$(SED) 's/\(SUBDIRS\s*[+:]\s*=\)/#\1/' $(PPPD_DIR)/pppd/plugins/Makefile.linux
 	$(SED) 's/FILTER=y/#FILTER=y/' $(PPPD_DIR)/pppd/Makefile.linux
 	$(SED) 's,(INSTALL) -s,(INSTALL),' $(PPPD_DIR)/*/Makefile.linux
 	$(SED) 's,(INSTALL) -s,(INSTALL),' $(PPPD_DIR)/pppd/plugins/*/Makefile.linux
 	$(SED) 's/ -o root//' $(PPPD_DIR)/*/Makefile.linux
 	$(SED) 's/ -g daemon//' $(PPPD_DIR)/*/Makefile.linux
+	echo "2011-08-01 gc: patch: apply debian's pppd patch (for GPRS DNS...)"
+	cd $(PPPD_DIR); zcat $(DL_DIR)/ppp_2.4.5-5.diff.gz | patch -p1
 	touch $@
 
 $(PPPD_DIR)/.configured: $(PPPD_DIR)/.unpacked
diff -Nru '--exclude=*~' buildroot-2.0-org/package/stunnel/stunnel.mk buildroot-2.0/package/stunnel/stunnel.mk
--- buildroot-2.0-org/package/stunnel/stunnel.mk	2008-12-08 09:15:27.000000000 +0100
+++ buildroot-2.0/package/stunnel/stunnel.mk	2011-08-01 14:45:53.000000000 +0200
@@ -3,7 +3,7 @@
 # stunnel
 #
 #############################################################
-STUNNEL_VERSION:=4.16
+STUNNEL_VERSION:=4.26
 STUNNEL_SOURCE:=stunnel-$(STUNNEL_VERSION).tar.gz
 STUNNEL_SITE:=http://www.stunnel.org/download/stunnel/src
 STUNNEL_CAT:=$(ZCAT)
@@ -56,7 +56,7 @@
 $(TARGET_DIR)/usr/bin/stunnel: $(STUNNEL_DIR)/src/stunnel
 	install -c $(STUNNEL_DIR)/src/stunnel $(TARGET_DIR)/usr/bin/stunnel
 	$(STRIPCMD) $(TARGET_DIR)/usr/bin/stunnel > /dev/null 2>&1
-ifeq ($(BR2_CROSS_TOOLCHAIN_TARGET_UTILS),y)
+ifeq ($(strip $(BR2_CROSS_TOOLCHAIN_TARGET_UTILS)),y)
 	mkdir -p $(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/target_utils
 	install -c $(TARGET_DIR)/usr/bin/stunnel \
 		$(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/target_utils/stunnel
@@ -77,6 +77,6 @@
 # Toplevel Makefile options
 #
 #############################################################
-ifeq ($(BR2_PACKAGE_STUNNEL),y)
+ifeq ($(strip $(BR2_PACKAGE_STUNNEL)),y)
 TARGETS+=stunnel
 endif
