diff -Nrub buildroot-2.0/Makefile new/Makefile
--- buildroot-2.0/Makefile	2008-11-07 09:57:59.000000000 +0100
+++ new/Makefile	2008-12-11 14:47:39.000000000 +0100
@@ -254,6 +254,7 @@
 endif
 
 include package/*/*.mk
+include weiss/Makefile
 
 TARGETS+=target-devfiles
 
diff -Nrub buildroot-2.0/package/Config.in new/package/Config.in
--- buildroot-2.0/package/Config.in	2008-12-11 01:33:30.000000000 +0100
+++ new/package/Config.in	2008-12-11 14:46:36.000000000 +0100
@@ -430,4 +430,6 @@
 # various games packages
 source "package/games/Config.in"
 
+source "weiss/package/Config.in"
+
 endmenu
diff -Nrub buildroot-2.0/package/lighttpd/lighttpd.mk new/package/lighttpd/lighttpd.mk
--- buildroot-2.0/package/lighttpd/lighttpd.mk	2008-12-08 09:15:27.000000000 +0100
+++ new/package/lighttpd/lighttpd.mk	2008-12-11 14:47:13.000000000 +0100
@@ -8,7 +8,7 @@
 LIGHTTPD_SITE:=http://www.lighttpd.net/download
 LIGHTTPD_INSTALL_STAGING = NO
 LIGHTTPD_INSTALL_TARGET = YES
-LIGHTTPD_DEPENDENCIES = uclibc
+LIGHTTPD_DEPENDENCIES = uclibc zlib
 LIGHTTPD_CONF_ENV =
 
 ifneq ($(BR2_LARGEFILE),y)
@@ -20,6 +20,7 @@
 	--libexecdir=/usr/lib \
 	--localstatedir=/var \
 	--program-prefix="" \
+	--without-bzip2 \
 	$(DISABLE_IPV6) \
 	$(LIGHTTPD_LFS)
 
diff -Nrub buildroot-2.0/package/mtd/mtd-utils/mtd.mk new/package/mtd/mtd-utils/mtd.mk
--- buildroot-2.0/package/mtd/mtd-utils/mtd.mk	2008-12-08 09:15:27.000000000 +0100
+++ new/package/mtd/mtd-utils/mtd.mk	2008-12-11 14:46:03.000000000 +0100
@@ -38,6 +38,7 @@
 	CC="$(HOSTCC)" CROSS= CFLAGS=-I$(LINUX_HEADERS_DIR)/include \
 		$(MAKE) LINUXDIR=$(LINUX_DIR) \
 		BUILDDIR=$(MTD_HOST_DIR) \
+		WITHOUT_XATTR=1 \
 		-C $(MTD_HOST_DIR) mkfs.jffs2
 
 $(SUMTOOL): $(MTD_HOST_DIR)/.unpacked
diff -Nrub buildroot-2.0/package/stunnel/stunnel.mk new/package/stunnel/stunnel.mk
--- buildroot-2.0/package/stunnel/stunnel.mk	2008-12-08 09:15:27.000000000 +0100
+++ new/package/stunnel/stunnel.mk	2008-12-15 15:29:04.000000000 +0100
@@ -3,7 +3,7 @@
 # stunnel
 #
 #############################################################
-STUNNEL_VERSION:=4.16
+STUNNEL_VERSION:=4.26
 STUNNEL_SOURCE:=stunnel-$(STUNNEL_VERSION).tar.gz
 STUNNEL_SITE:=http://www.stunnel.org/download/stunnel/src
 STUNNEL_CAT:=$(ZCAT)
@@ -56,7 +56,7 @@
 $(TARGET_DIR)/usr/bin/stunnel: $(STUNNEL_DIR)/src/stunnel
 	install -c $(STUNNEL_DIR)/src/stunnel $(TARGET_DIR)/usr/bin/stunnel
 	$(STRIPCMD) $(TARGET_DIR)/usr/bin/stunnel > /dev/null 2>&1
-ifeq ($(BR2_CROSS_TOOLCHAIN_TARGET_UTILS),y)
+ifeq ($(strip $(BR2_CROSS_TOOLCHAIN_TARGET_UTILS)),y)
 	mkdir -p $(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/target_utils
 	install -c $(TARGET_DIR)/usr/bin/stunnel \
 		$(STAGING_DIR)/$(REAL_GNU_TARGET_NAME)/target_utils/stunnel
@@ -77,6 +77,6 @@
 # Toplevel Makefile options
 #
 #############################################################
-ifeq ($(BR2_PACKAGE_STUNNEL),y)
+ifeq ($(strip $(BR2_PACKAGE_STUNNEL)),y)
 TARGETS+=stunnel
 endif
diff -Nrub buildroot-2.0/package/ntp/ntp.mk new/package/ntp/ntp.mk
--- buildroot-2.0/package/ntp/ntp.mk	2009-07-24 09:32:30.216504093 +0200
+++ new/package/ntp/ntp.mk	2009-07-24 09:31:56.601783992 +0200
@@ -37,6 +37,9 @@
 		--target=$(GNU_TARGET_NAME) \
 		--host=$(GNU_TARGET_NAME) \
 		--build=$(GNU_HOST_NAME) \
+		--enable-MEINBERG \
+		--enable-RAWDCF \
+		--enable-parse-clocks \
 		--prefix=/usr \
 		--exec-prefix=/usr \
 		--bindir=/usr/bin \
