--- comgt.0.32/Makefile	2007-11-27 18:13:03.000000000 +0100
+++ comgt.0.32-new/Makefile	2007-11-27 18:10:24.000000000 +0100
@@ -23,8 +23,6 @@
 #
 #
 
-LIB     = -L/usr/local/lib
-INC     = -I/usr/local/include
 EXE	= /usr/local/bin
 MAN     = /usr/share/man/man1
 CPROG	= comgt
@@ -33,9 +31,6 @@
 BIN     = $(CPROG) 
 MANP	= comgt.1 sigmon.1
 
-CFLAGS  = -c
-LDFLAGS =
-
 all: $(BIN)
 
 install:
@@ -72,8 +67,8 @@
 
 
 comgt: comgt.o
-	cc comgt.o $(LDFLAGS) -o comgt
+	$(CC) comgt.o $(LDFLAGS) -o comgt
 
 comgt.o: comgt.c comgt.h
-	cc comgt.c $(CFLAGS) 
+	$(CC) -c comgt.c $(CFLAGS) 
 
