diff -Nrub --exclude '*~' openvpn-2.3.2-org//src/openvpn/init.c openvpn-2.3.2/src/openvpn/init.c
--- openvpn-2.3.2-org//src/openvpn/init.c	2013-05-31 14:00:25.000000000 +0200
+++ openvpn-2.3.2/src/openvpn/init.c	2013-06-18 10:40:02.000000000 +0200
@@ -660,7 +660,9 @@
   {
     struct route_gateway_info rgi;
     get_default_gateway(&rgi);
+#ifndef ENABLE_SMALL
     print_default_gateway(M_INFO, &rgi);
+#endif
     return false;
   }
 #endif
diff -Nrub --exclude '*~' openvpn-2.3.2-org//src/openvpn/options.c openvpn-2.3.2/src/openvpn/options.c
--- openvpn-2.3.2-org//src/openvpn/options.c	2013-05-31 14:00:25.000000000 +0200
+++ openvpn-2.3.2/src/openvpn/options.c	2013-06-18 10:40:10.000000000 +0200
@@ -4065,7 +4065,9 @@
       struct route_gateway_info rgi;
       VERIFY_PERMISSION (OPT_P_GENERAL);
       get_default_gateway(&rgi);
+#ifndef ENABLE_SMALL
       print_default_gateway(M_INFO, &rgi);
+#endif
       openvpn_exit (OPENVPN_EXIT_STATUS_GOOD); /* exit point */
     }
 #endif
diff -Nrub --exclude '*~' openvpn-2.3.2-org//src/openvpn/route.c openvpn-2.3.2/src/openvpn/route.c
--- openvpn-2.3.2-org//src/openvpn/route.c	2013-05-31 14:00:25.000000000 +0200
+++ openvpn-2.3.2/src/openvpn/route.c	2013-06-18 10:39:36.000000000 +0200
@@ -596,8 +596,10 @@
     {
       setenv_route_addr (es, "net_gateway", rl->rgi.gateway.addr, -1);
 #ifdef ENABLE_DEBUG
+#ifndef ENABLE_SMALL
       print_default_gateway (D_ROUTE, &rl->rgi);
 #endif
+#endif
     }
   else
     {
